src/provider.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import * as React from 'react';
+import { checkSessionAction } from './actions.js';
+
+export const Provider = ({ children }: React.PropsWithChildren) => {
```

Comment:
We should rename this to be more specific, like `AuthKitProvider` or `AuthProvider`.